x-pack/plugins/reporting/export_types/printable_pdf/server/lib/layouts/layout.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Size } from '../../../../../types';
+import { ViewZoomWidthHeight } from './index.d';
+
+export interface PageSizeParams {
+  pageMarginTop: number;
+  pageMarginBottom: number;
+  pageMarginWidth: number;
+  tableBorderWidth: number;
+  headingHeight: number;
+  subheadingHeight: number;
+}
+
+export interface PdfImageSize {
+  width: number;
+  height?: number;
+}
+
+export abstract class Layout {
+  public id: string = '';
+
+  constructor(id: string) {
+    this.id = id;
+  }
+
+  public abstract getPdfImageSize(): PdfImageSize;
+
+  public abstract getPdfPageOrientation(): string | undefined;
+
+  public abstract getPdfPageSize(pageSizeParamsIn: PageSizeParams): string | Size;
```

Comment:
I would just call is `pageSizeParams` and drop the `In` part of the name. Not a common convention in our code base to suffix input parameters with `In`.